front/lib/resources/run_resource.ts

```diff
@@ -108,7 +109,19 @@ export class RunResource extends BaseResource<RunModel> {
   }
 
   static async deleteAllByAppId(appId: ModelId, transaction?: Transaction) {
-    // TODO: First, delete the run_usage before deleting the run.
+    assert(typeof appId === "number");
+    await RunUsageModel.destroy({
+      where: {
+        runId: {
+          [Op.in]: Sequelize.literal(
+            // Sequelize prevents other safer constructs due to typing with the destroy method.
+            // `appId` cannot cannot be user provided + assert above.
+            `(SELECT id FROM runs WHERE appId = '${appId}')`
```

Comment:
Given that we never delete runs in front (IIRC) this might be crazy big on some workspaces. Can we do proper pagination instead and delete in batches? 

Otherwise, this will lock the transaction for a pretty long time!

Reply:
I'm in favor of nuking that transaction I think it does not make sense?

Reply:
👍